runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java

```diff
@@ -70,17 +76,43 @@
 	@Inject
 	private SkillIconManager skillIconManager;
 
-	private NavigationButton navButton;
-	private XpPanel xpPanel;
+	@Inject
+	private XpTrackerConfig xpTrackerConfig;
 
-	private final XpState xpState = new XpState();
+	@Inject
+	private XpState xpState;
 
+	@Inject
+	private XpPauseState xpPauseState;
+
+	private NavigationButton navButton;
+	private XpPanel xpPanel;
 	private WorldResult worlds;
 	private XpWorldType lastWorldType;
 	private String lastUsername;
+	private Instant lastTick;
 
 	private final XpClient xpClient = new XpClient();
 
+
+	@Provides
+	XpTrackerConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(XpTrackerConfig.class);
+	}
+
+	@Provides
+	XpState provideInternalState()
```

Comment:
Why are you using provider for this when it has no dependencies? You can just construct it and send it around

Reply:
I'll simplify these so construction time requirements is no longer necessary.